src/main/java/Duke.java

```diff
@@ -1,10 +1,108 @@
+import java.util.Scanner;
+
 public class Duke {
+
+    private static final String LINE = "    __________________________________________________________________________";
+    private static final String INDENT = "     ";
+
+    private static boolean isClosed = false;
+    private static TaskList tasks = new TaskList();
+    private static int counter = 0;
+
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
-    }
-}
+        greetingMessage();
+        Scanner sc = new Scanner(System.in);
+        while (!isClosed) {
+            try {
+                String command = sc.nextLine();
+                if (command.equals("bye")) {
+                    bye();
+                } else if (command.equals("list")) {
+                    printList();
+                } else {
+                    if (command.startsWith("mark")) {
+                        String[] splitted = command.split("\\s+");
+                        Integer index;
+                        if (splitted.length == 1) {
+                            throw new DukeUnspecifiedNumberException("mark");
+                        }
+                        try {
+                            index = Integer.valueOf(splitted[1]);
+                            tasks.markAsDone(index);
+                        } catch (NumberFormatException err) {
+                            throw new DukeInvalidNumberFormatException("mark");
+                        }
```

Comment:
Is it possible to avoid deep nesting with more than 3 levels of indentation?

Reply:
Yup, I just caught up on the SLAP method and the arrowhead style, I would consider implementing it in the next few iterations. Thank you for your comment.